path_login.go

```diff
@@ -222,129 +214,100 @@ func (b *kubeAuthBackend) aliasLookahead(ctx context.Context, req *logical.Reque
 	}, nil
 }
 
+// parseAndVerifySignature parses the JWT token validating the signature against
+// any of the keys passed in.
+func (b *kubeAuthBackend) parseAndVerifySignature(token string, keys ...interface{}) (*jwt.Token, error) {
+	for i, k := range keys {
+		result, err := jwt.Parse(token, func(t *jwt.Token) (interface{}, error) {
```

Comment:
:+1: I was initially thinking I would avoid hard-coding which methods we supported since JWT could be updated to support, e.g., EdDSA, but you're right, it is more important to guard against `"alg"` attacks.

I'll add verification of the signing method and the keys themselves to match the old behavior better.